--- src/err.cpp
+++ src/err.cpp
@@ -31,6 +31,8 @@
 #include "err.hpp"
 #include "macros.hpp"
 
+#include <asl.h>
+
 const char *zmq::errno_to_string (int errno_)
 {
     switch (errno_) {
@@ -85,7 +87,11 @@
 #else
     LIBZMQ_UNUSED (errmsg_);
     print_backtrace ();
-    abort ();
+    // abort ();
+    aslmsg msg = asl_new(ASL_TYPE_MSG);
+    asl_set(msg, ASL_KEY_READ_UID, "-1");
+    asl_log(NULL, msg, ASL_LEVEL_WARNING, "zmq_abort() skipped, with message: %s", errmsg_);
+    asl_free(msg);
 #endif
 }
 
