--- src/kqueue.cpp
+++ src/kqueue.cpp
@@ -172,6 +172,14 @@
 
 void zmq::kqueue_t::loop ()
 {
+    // Name this thread.
+    {
+        char *threadName = nullptr;
+        asprintf(&threadName, "org.zeromq.kqueue  fd=%d", kqueue_fd);
+        pthread_setname_np(threadName);
+        free(threadName);
+    }
+
     while (true) {
         //  Execute any due timers.
         int timeout = (int) execute_timers ();
